### PR TITLE
Upgrade to ROS 2 Eloquent

### PR DESCRIPTION
**Pull request by henningkayser**
This PR applies necessary `CMakeLists.txt` fixes to to compile `moveit_core` and `moveit_ros_planning` with ROS 2 Eloquent. CI is still running on dashing. Once PR #153 is merged we should switch ROS_DISTRO to eloquent in Travis. Since these changes also work for dashing feel free to merge this right away.

### CODE REVIEW

**Comment by henningkayser** [comment]
@JafarAbdi @RoboticsYY @nbbrooks This branch compiles on eloquent (simply install `ros-eloquent-desktop` and follow the instructions in the README). Please start rebasing the open PRs on to this one and see if it works for you. Keep in mind that with eloquent STREAM/THROTTLE macros are supported from now on.

**Comment by henningkayser** [comment]
> This seems to move dependencies from `ament_target_dependencies` to `target_link_libraries`, while the latter is not recommended as it doesn't order dependencies correctly (see [here](https://index.ros.org/doc/ros2/Tutorials/Ament-CMake-Documentation/#adding-dependencies)).
> Can you motivate why this is necessary?

`ament_target_dependencies()` is only meant to be used for package names, not for library targets. In dashing there was a bug where non-packages where accepted, now this causes CMake to fail. See https://github.com/ament/ament_cmake/pull/180

File: moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,6 @@ ament_target_dependencies(${MOVEIT_LIB_NAME}
     rclcpp
     pluginlib
     moveit_core
-    moveit_planning_scene
```
Comment (nbbrooks) [inline moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt]: don't see a target_link_libraries for this - was it not needed?

File: moveit_core/collision_distance_field/CMakeLists.txt [hunk 2]
```diff
@@ -12,9 +12,8 @@ ament_target_dependencies(${MOVEIT_LIB_NAME}
   urdf
   visualization_msgs
   tf2_eigen
-  angles
```
[the remark this answers is not in the document]
Comment (henningkayser) [reply moveit_core/collision_distance_field/CMakeLists.txt]: I checked the code and apparently it isn't

File: moveit_core/collision_detection_fcl/CMakeLists.txt [hunk 3]
```diff
@@ -28,7 +27,10 @@ ament_target_dependencies(collision_detector_fcl_plugin
   visualization_msgs
   pluginlib
   rmw_implementation
-  ${MOVEIT_LIB_NAME})
+)
```
[the remark this answers is not in the document]
Comment (henningkayser) [reply moveit_core/collision_detection_fcl/CMakeLists.txt]: Yes, `ament_target_dependencies` is meant to be used for packages only, libraries need to be linked via `target_link_libraries` (see comment above)

File: moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt (hunk 1 above)
Comment (henningkayser) [reply moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt]: `moveit_planning_scene` is a library included in the package `moveit_core` which is already included.

**Comment by anthonytec2** [comment]
I have been trying to get the above to work on my own ROS2 Eloquent docker container, although I am also receiving an error on trying to find libtinyxml, has anyone seen this before?

File: moveit_core/collision_detection_fcl/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (nbbrooks) [reply moveit_core/collision_detection_fcl/CMakeLists.txt]: OK - there are some files in this PR that still have

`ament_target_dependencies(${MOVEIT_LIB_NAME}`

do they also need to be removed?

**Comment by RoboticsYY** [comment]
Looks good to me! I think [geometric_shapes PR#124](https://github.com/ros-planning/geometric_shapes/pull/124) and [srdf PR#4](https://github.com/PickNikRobotics/srdfdom/pull/4) need to be merged before CI pass.

File: moveit_core/collision_distance_field/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (davetcoleman) [reply moveit_core/collision_distance_field/CMakeLists.txt]: Why was it added to the root moveit_core CMakeLists then?

File: moveit_core/collision_distance_field/CMakeLists.txt [hunk 4]
```diff
@@ -12,9 +12,8 @@ ament_target_dependencies(${MOVEIT_LIB_NAME}
   urdf
```
Comment (nbbrooks) [inline moveit_core/collision_distance_field/CMakeLists.txt]: ex: this file still has "ament_target_dependencies(${MOVEIT_LIB_NAME}"

File: moveit_core/constraint_samplers/CMakeLists.txt [hunk 5]
```diff
@@ -8,22 +8,18 @@ add_library(${MOVEIT_LIB_NAME} SHARED
   src/union_constraint_sampler.cpp
 )
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-
 ament_target_dependencies(${MOVEIT_LIB_NAME}
```
Comment (nbbrooks) [inline moveit_core/constraint_samplers/CMakeLists.txt]: remove MOVEIT_LIB_NAME

Comment (henningkayser) [reply moveit_core/constraint_samplers/CMakeLists.txt]: MOVEIT_LIB_NAME is the target we are adding the dependencies for, not a dependency itself.

File: moveit_core/collision_distance_field/CMakeLists.txt (hunk 4 above)
Comment (henningkayser) [reply moveit_core/collision_distance_field/CMakeLists.txt]: same

File: moveit_core/constraint_samplers/CMakeLists.txt (hunk 5 above)
Comment (nbbrooks) [reply moveit_core/constraint_samplers/CMakeLists.txt]: Got it, thanks

**Pull request merged**